taurus/entity/template/base_template.py

```diff
@@ -66,4 +66,4 @@ def _homogenize_ranges(template_or_tuple):
                     return [first, second]
                 else:
                     raise ValueError("Range and template are inconsistent")
-        raise ValueError("Expected a template or (template, bounds) tuple")
+        raise TypeError("Expected a template or (template, bounds) tuple")  # pragma: no cover
```

Comment:
This line cannot be covered naturally because the methods that call `_homogenize_ranges` already check that the argument they are passing is an `AttributeTemplate`, `list`, or `tuple`